analysis/config/dataflow_config.go

```diff
@@ -238,6 +238,12 @@ func (ss SlicingSpec) IsBacktracePoint(cid CodeIdentifier) bool {
 	return ExistsCid(ss.BacktracePoints, cid.equalOnNonEmptyFields)
 }
 
+// ExceedsMaxDepth returns true if the input exceeds the maximum depth parameter of the configuration.
+// (this implements the logic for using maximum depth; if the configuration setting is < 0, then this returns false)
+func (ss SlicingSpec) ExceedsMaxDepth(d int) bool {
+	return !(ss.UnsafeMaxDepth <= 0) && d > ss.UnsafeMaxDepth
```

Comment:
nit: `ss.UnsafeMaxDepth > 0 && d > ss.UnsafeMaxDepth` 
But I'm not sure why this function is needed on top of the config one, whose option should be overridden as necessary.

Reply:
Oh good point, I forgot that the option gets overridden